File: archive/feature/test/src/main/java/com/jian/nemo/feature/test/presentation/settings/components/TestSettingsDialogs.kt
package com.jian.nemo.feature.test.presentation.settings.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.ui.graphics.Color

/**
 * 测试设置对话框组件
 *
 * 提取自：TestSettingsScreen.kt 行139-235
 */

/**
 * 自定义题目数量对话框
 *
 * @param show 是否显示
 * @param initialValue 初始值
 * @param onDismiss 关闭回调
 * @param onConfirm 确认回调，参数为用户输入的数量
 */
@OptIn(ExperimentalLayoutApi::class)
@Composable
fun CustomQuestionCountDialog(
    show: Boolean,
    initialValue: Int,
    onDismiss: () -> Unit,
    onConfirm: (Int) -> Unit
) {
    if (!show) return

    // 内部状态
    var currentValue by remember(show) { mutableIntStateOf(initialValue) }
    
    // 预设选项
    val presets = remember { listOf(10, 15, 20, 25, 30, 50) }

    androidx.compose.ui.window.Dialog(
        onDismissRequest = onDismiss
    ) {
        val isDark = isSystemInDarkTheme()
        Surface(
            shape = RoundedCornerShape(28.dp),
            color = if (isDark) MaterialTheme.colorScheme.surface else Color.White,
            tonalElevation = 0.dp, // Remove tonal/gradient overlay
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            shadowElevation = 8.dp
        ) {
            Column(
                modifier = Modifier.padding(24.dp),
                horizontalAlignment = androidx.compose.ui.Alignment.CenterHorizontally
            ) {
                // 1. 标题
                Text(
                    text = "自定义题数",
                    style = MaterialTheme.typography.headlineSmall,
                    fontWeight = androidx.compose.ui.text.font.FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                Text(
                    text = "设置本次测试的题目数量",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(top = 8.dp, bottom = 24.dp)
                )

                // 2. 核心交互区域 ( -  数值  + )
                Row(
                    verticalAlignment = androidx.compose.ui.Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.Center,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    // 减号按钮
                    FilledIconButton(
                        onClick = { if (currentValue > 1) currentValue-- },
                        enabled = currentValue > 1,
                        colors = IconButtonDefaults.filledIconButtonColors(
                            containerColor = MaterialTheme.colorScheme.secondaryContainer,
                            contentColor = MaterialTheme.colorScheme.onSecondaryContainer
                        ),
                        modifier = Modifier.size(48.dp)
                    ) {
                        Text("-", style = MaterialTheme.typography.titleLarge)
                    }

                    // 数值显示
                    Text(
                        text = "$currentValue",
                        style = MaterialTheme.typography.displayMedium,
                        fontWeight = androidx.compose.ui.text.font.FontWeight.Bold,
                        color = MaterialTheme.colorScheme.primary,
                        modifier = Modifier
                            .padding(horizontal = 24.dp)
                            .widthIn(min = 64.dp), // 防止数字变动导致抖动
                        textAlign = androidx.compose.ui.text.style.TextAlign.Center
                    )

                    // 加号按钮
                    FilledIconButton(
                        onClick = { if (currentValue < 1000) currentValue++ },
                        enabled = currentValue < 1000,
                        colors = IconButtonDefaults.filledIconButtonColors(
                            containerColor = MaterialTheme.colorScheme.secondaryContainer,
                            contentColor = MaterialTheme.colorScheme.onSecondaryContainer
                        ),
                        modifier = Modifier.size(48.dp)
                    ) {
                        Text("+", style = MaterialTheme.typography.titleLarge)
                    }
                }
                
                Text(
                    text = "题",
                    style = MaterialTheme.typography.labelLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(top = 4.dp, bottom = 24.dp)
                )

                // 3. 快速预设 (Chips)
                FlowRow(
                    horizontalArrangement = Arrangement.Center,
                    verticalArrangement = Arrangement.spacedBy(8.dp),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    presets.forEach { preset ->
                        FilterChip(
                            selected = currentValue == preset,
                            onClick = { currentValue = preset },
                            label = { Text("${preset}") },
                            colors = FilterChipDefaults.filterChipColors(
                                selectedContainerColor = MaterialTheme.colorScheme.primaryContainer,
                                selectedLabelColor = MaterialTheme.colorScheme.onPrimaryContainer
                            ),
                            border = FilterChipDefaults.filterChipBorder(
                                enabled = true,
                                selected = currentValue == preset,
                                borderColor = MaterialTheme.colorScheme.outline.copy(alpha = 0.5f)
                            )
                        )
                        Spacer(modifier = Modifier.width(8.dp))
                    }
                }

                Spacer(modifier = Modifier.height(32.dp))

                // 4. 底部操作按钮
                Row(
                   modifier = Modifier.fillMaxWidth()
                ) {
                    // 取消按钮
                    TextButton(
                        onClick = onDismiss,
                        modifier = Modifier
                            .weight(1f)
                            .height(48.dp),
                        shape = RoundedCornerShape(24.dp)
                    ) {
                        Text("取消", style = MaterialTheme.typography.titleMedium)
                    }
                    
                    Spacer(modifier = Modifier.width(12.dp))
                    
                    // 确认按钮
                    Button(
                        onClick = {
                            if (currentValue in 1..1000) {
                                onConfirm(currentValue)
                                onDismiss()
                            }
                        },
                        modifier = Modifier
                            .weight(1f)
                            .height(48.dp),
                        shape = RoundedCornerShape(24.dp), // Fully rounded pill shape
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MaterialTheme.colorScheme.primary,
                            contentColor = MaterialTheme.colorScheme.onPrimary
                        ),
                        elevation = ButtonDefaults.buttonElevation(defaultElevation = 2.dp)
                    ) {
                        Text("确定", style = MaterialTheme.typography.titleMedium, fontWeight = androidx.compose.ui.text.font.FontWeight.Bold)
                    }
                }
            }
        }
    }
}

/**
 * 自定义时间限制对话框 - UI/UX Pro Max
 *
 * @param show 是否显示
 * @param initialValue 初始值（分钟）
 * @param onDismiss 关闭回调
 * @param onConfirm 确认回调，参数为用户输入的分钟数
 */
@OptIn(ExperimentalLayoutApi::class)
@Composable
fun CustomTimeLimitDialog(
    show: Boolean,
    initialValue: Int,
    onDismiss: () -> Unit,
    onConfirm: (Int) -> Unit
) {
    if (!show) return

    // 内部状态
    var currentValue by remember(show) { mutableIntStateOf(initialValue) }
    
    // 预设选项
    val presets = remember { listOf(5, 10, 20, 30, 45, 60) }

    androidx.compose.ui.window.Dialog(
        onDismissRequest = onDismiss
    ) {
        val isDark = isSystemInDarkTheme()
        Surface(
            shape = RoundedCornerShape(28.dp),
            color = if (isDark) MaterialTheme.colorScheme.surface else Color.White,
            tonalElevation = 0.dp, // Remove tonal/gradient overlay
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            shadowElevation = 8.dp
        ) {
            Column(
                modifier = Modifier.padding(24.dp),
                horizontalAlignment = androidx.compose.ui.Alignment.CenterHorizontally
            ) {
                // 1. 标题
                Text(
                    text = "自定义时长",
                    style = MaterialTheme.typography.headlineSmall,
                    fontWeight = androidx.compose.ui.text.font.FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                Text(
                    text = "设置测试时间限制 (分钟)",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(top = 8.dp, bottom = 24.dp)
                )

                // 2. 核心交互区域 ( -  数值  + )
                Row(
                    verticalAlignment = androidx.compose.ui.Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.Center,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    // 减号按钮
                    FilledIconButton(
                        onClick = { if (currentValue > 0) currentValue-- },
                        enabled = currentValue > 0,
                        colors = IconButtonDefaults.filledIconButtonColors(
                            containerColor = MaterialTheme.colorScheme.secondaryContainer,
                            contentColor = MaterialTheme.colorScheme.onSecondaryContainer
                        ),
                        modifier = Modifier.size(48.dp)
                    ) {
                        Text("-", style = MaterialTheme.typography.titleLarge)
                    }

                    // 数值显示
                    Text(
                        text = "$currentValue",
                        style = MaterialTheme.typography.displayMedium,
                        fontWeight = androidx.compose.ui.text.font.FontWeight.Bold,
                        color = MaterialTheme.colorScheme.primary,
                        modifier = Modifier
                            .padding(horizontal = 24.dp)
                            .widthIn(min = 64.dp), // 防止数字变动导致抖动
                        textAlign = androidx.compose.ui.text.style.TextAlign.Center
                    )

                    // 加号按钮
                    FilledIconButton(
                        onClick = { if (currentValue < 180) currentValue++ },
                        enabled = currentValue < 180,
                        colors = IconButtonDefaults.filledIconButtonColors(
                            containerColor = MaterialTheme.colorScheme.secondaryContainer,
                            contentColor = MaterialTheme.colorScheme.onSecondaryContainer
                        ),
                        modifier = Modifier.size(48.dp)
                    ) {
                        Text("+", style = MaterialTheme.typography.titleLarge)
                    }
                }
                
                Text(
                    text = if (currentValue == 0) "不限时间" else "分钟",
                    style = MaterialTheme.typography.labelLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(top = 4.dp, bottom = 24.dp)
                )

                // 3. 快速预设 (Chips)
                FlowRow(
                    horizontalArrangement = Arrangement.Center,
                    verticalArrangement = Arrangement.spacedBy(8.dp),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    presets.forEach { preset ->
                        FilterChip(
                            selected = currentValue == preset,
                            onClick = { currentValue = preset },
                            label = { Text("${preset}m") },
                            colors = FilterChipDefaults.filterChipColors(
                                selectedContainerColor = MaterialTheme.colorScheme.primaryContainer,
                                selectedLabelColor = MaterialTheme.colorScheme.onPrimaryContainer
                            ),
                            border = FilterChipDefaults.filterChipBorder(
                                enabled = true,
                                selected = currentValue == preset,
                                borderColor = MaterialTheme.colorScheme.outline.copy(alpha = 0.5f)
                            )
                        )
                        Spacer(modifier = Modifier.width(8.dp))
                    }
                }

                Spacer(modifier = Modifier.height(32.dp))

                // 4. 底部操作按钮
                Row(
                   modifier = Modifier.fillMaxWidth()
                ) {
                    // 取消按钮
                    TextButton(
                        onClick = onDismiss,
                        modifier = Modifier
                            .weight(1f)
                            .height(48.dp),
                        shape = RoundedCornerShape(24.dp)
                    ) {
                        Text("取消", style = MaterialTheme.typography.titleMedium)
                    }
                    
                    Spacer(modifier = Modifier.width(12.dp))
                    
                    // 确认按钮
                    Button(
                        onClick = {
                            if (currentValue in 0..180) {
                                onConfirm(currentValue)
                                onDismiss()
                            }
                        },
                        modifier = Modifier
                            .weight(1f)
                            .height(48.dp),
                        shape = RoundedCornerShape(24.dp), // Fully rounded pill shape
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MaterialTheme.colorScheme.primary,
                            contentColor = MaterialTheme.colorScheme.onPrimary
                        ),
                        elevation = ButtonDefaults.buttonElevation(defaultElevation = 2.dp)
                    ) {
                        Text("确定", style = MaterialTheme.typography.titleMedium, fontWeight = androidx.compose.ui.text.font.FontWeight.Bold)
                    }
                }
            }
        }
    }
}

File: packages/features/feature_test/lib/src/test/components/test_settings_dialogs.dart
import 'package:flutter/material.dart';
import 'package:flutter_hooks/flutter_hooks.dart';

class CustomQuestionCountDialog extends HookWidget {
  final bool show;
  final int initialValue;
  final VoidCallback onDismiss;
  final ValueChanged<int> onConfirm;

  const CustomQuestionCountDialog({
    super.key,
    required this.show,
    required this.initialValue,
    required this.onDismiss,
    required this.onConfirm,
  });

  @override
  Widget build(BuildContext context) {
    if (!show) return const SizedBox.shrink();

    final currentValue = useState(initialValue);
    final presets = [10, 15, 20, 25, 30, 50];

    return Dialog(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(28)),
      elevation: 0,
      backgroundColor: Colors.transparent,
      child: _DialogContainer(
        title: "自定义题数",
        subtitle: "设置本次测试的题目数量",
        currentValue: currentValue.value,
        minValue: 1,
        maxValue: 1000,
        unit: "题",
        presets: presets,
        onValueChange: (val) => currentValue.value = val,
        onDismiss: onDismiss,
        onConfirm: () {
          onConfirm(currentValue.value);
          onDismiss();
        },
      ),
    );
  }
}

class CustomTimeLimitDialog extends HookWidget {
  final bool show;
  final int initialValue;
  final VoidCallback onDismiss;
  final ValueChanged<int> onConfirm;

  const CustomTimeLimitDialog({
    super.key,
    required this.show,
    required this.initialValue,
    required this.onDismiss,
    required this.onConfirm,
  });

  @override
  Widget build(BuildContext context) {
    if (!show) return const SizedBox.shrink();

    final currentValue = useState(initialValue);
    final presets = [5, 10, 20, 30, 45, 60];

    return Dialog(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(28)),
      elevation: 0,
      backgroundColor: Colors.transparent,
      child: _DialogContainer(
        title: "自定义时长",
        subtitle: "设置测试时间限制 (分钟)",
        currentValue: currentValue.value,
        minValue: 0,
        maxValue: 180,
        unit: currentValue.value == 0 ? "不限时间" : "分钟",
        presets: presets,
        onValueChange: (val) => currentValue.value = val,
        onDismiss: onDismiss,
        onConfirm: () {
          onConfirm(currentValue.value);
          onDismiss();
        },
      ),
    );
  }
}

class _DialogContainer extends StatelessWidget {
  final String title;
  final String subtitle;
  final int currentValue;
  final int minValue;
  final int maxValue;
  final String unit;
  final List<int> presets;
  final ValueChanged<int> onValueChange;
  final VoidCallback onDismiss;
  final VoidCallback onConfirm;

  const _DialogContainer({
    required this.title,
    required this.subtitle,
    required this.currentValue,
    required this.minValue,
    required this.maxValue,
    required this.unit,
    required this.presets,
    required this.onValueChange,
    required this.onDismiss,
    required this.onConfirm,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final isDark = theme.brightness == Brightness.dark;

    return Container(
      padding: const EdgeInsets.all(24),
      decoration: BoxDecoration(
        color: isDark ? theme.colorScheme.surface : Colors.white,
        borderRadius: BorderRadius.circular(28),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withValues(alpha: 0.1),
            blurRadius: 10,
            offset: const Offset(0, 4),
          ),
        ],
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Text(
            title,
            style: theme.textTheme.headlineSmall?.copyWith(
              fontWeight: FontWeight.bold,
              color: theme.colorScheme.onSurface,
            ),
          ),
          const SizedBox(height: 8),
          Text(
            subtitle,
            style: theme.textTheme.bodyMedium?.copyWith(
              color: theme.colorScheme.onSurfaceVariant,
            ),
            textAlign: TextAlign.center,
          ),
          const SizedBox(height: 24),
          Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              _IconButton(
                icon: Icons.remove,
                enabled: currentValue > minValue,
                onPressed: () => onValueChange(currentValue - 1),
              ),
              const SizedBox(width: 24),
              SizedBox(
                width: 80, // Prevent jitter
                child: Text(
                  "$currentValue",
                  style: theme.textTheme.displayMedium?.copyWith(
                    fontWeight: FontWeight.bold,
                    color: theme.colorScheme.primary,
                  ),
                  textAlign: TextAlign.center,
                ),
              ),
              const SizedBox(width: 24),
              _IconButton(
                icon: Icons.add,
                enabled: currentValue < maxValue,
                onPressed: () => onValueChange(currentValue + 1),
              ),
            ],
          ),
          const SizedBox(height: 4),
          Text(
            unit,
            style: theme.textTheme.labelLarge?.copyWith(
              color: theme.colorScheme.onSurfaceVariant,
            ),
          ),
          const SizedBox(height: 24),
          Wrap(
            spacing: 8,
            runSpacing: 8,
            alignment: WrapAlignment.center,
            children: presets.map((preset) {
              final isSelected = currentValue == preset;
              return ChoiceChip(
                label: Text(preset.toString()),
                selected: isSelected,
                onSelected: (_) => onValueChange(preset),
                selectedColor: theme.colorScheme.primaryContainer,
                labelStyle: TextStyle(
                  color: isSelected ? theme.colorScheme.onPrimaryContainer : theme.colorScheme.onSurface,
                ),
                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
              );
            }).toList(),
          ),
          const SizedBox(height: 32),
          Row(
            children: [
              Expanded(
                child: TextButton(
                  onPressed: onDismiss,
                  style: TextButton.styleFrom(
                    minimumSize: const Size(0, 48),
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24)),
                  ),
                  child: const Text("取消", style: TextStyle(fontSize: 16)),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: ElevatedButton(
                  onPressed: onConfirm,
                  style: ElevatedButton.styleFrom(
                    backgroundColor: theme.colorScheme.primary,
                    foregroundColor: theme.colorScheme.onPrimary,
                    minimumSize: const Size(0, 48),
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24)),
                    elevation: 2,
                  ),
                  child: const Text("确定", style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold)),
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
}

class _IconButton extends StatelessWidget {
  final IconData icon;
  final bool enabled;
  final VoidCallback onPressed;

  const _IconButton({
    required this.icon,
    required this.enabled,
    required this.onPressed,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Material(
      color: enabled ? theme.colorScheme.secondaryContainer : theme.colorScheme.surfaceContainerHighest.withValues(alpha: 0.5),
      borderRadius: BorderRadius.circular(12),
      child: InkWell(
        onTap: enabled ? onPressed : null,
        borderRadius: BorderRadius.circular(12),
        child: Container(
          width: 48,
          height: 48,
          alignment: Alignment.center,
          child: Icon(
            icon,
            color: enabled ? theme.colorScheme.onSecondaryContainer : theme.colorScheme.onSurfaceVariant.withValues(alpha: 0.3),
          ),
        ),
      ),
    );
  }
}
